File: FantasyRadio/src/main/java/ru/sigil/fantasyradio/dagger/Bootstrap.java
package ru.sigil.fantasyradio.dagger;

import ru.sigil.fantasyradio.MyApp;

/**
 * Created by NamelessOne
 * on 17.09.2016.
 */
public enum Bootstrap {
    INSTANCE;

    private PlayerComponent bootstrap;

    public void setup(MyApp app) {
        PlayerModule module = new PlayerModule(app);
        bootstrap =  DaggerPlayerComponent.builder().playerModule(module).build();
    }

    public PlayerComponent getBootstrap() {
        return bootstrap;
    }
}

File: FantasyRadio/src/main/kotlin/ru/sigil/fantasyradio/dagger/Bootstrap.kt
package ru.sigil.fantasyradio.dagger

import ru.sigil.fantasyradio.MyApp



/**
 * Created by namelessone
 * on 01.12.18.
 */
enum class Bootstrap {
    INSTANCE;

    private var bootstrap: PlayerComponent? = null

    fun setup(app: MyApp) {
        val module = PlayerModule(app)
        bootstrap = DaggerPlayerComponent.builder().playerModule(module).build()
    }

    fun getBootstrap(): PlayerComponent {
        return bootstrap!!
    }
}